playlistmanager.lua

```diff
@@ -237,6 +238,40 @@ if settings.system=="auto" then
   end
 end
 
+-- auto calculate showamount
+if settings.showamount=='auto' then
+  -- same as draw_playlist() height
+  local playlist_h = 360
+  if mp.get_property("osd-align-x") == "left" and mp.get_property("osd-align-y") == "top" then
+    -- both top and bottom with same padding
+    playlist_h = playlist_h - settings.text_padding_y * 2
+  end
+  
+  -- osd-font-size is based on 720p height
+  -- see https://mpv.io/manual/stable/#options-osd-font-size 
+  -- details in https://mpv.io/manual/stable/#options-sub-font-size
+  -- draw_playlist() is based on 360p height, so we need to divide by 2
+  local fs = mp.get_property_native('osd-font-size') / 2
+  -- get the ass font size
+  if settings.style_ass_tags ~= nil then
+    local ass_fs_tag = settings.style_ass_tags:match('\\fs%d+')
+    if ass_fs_tag ~= nil then
+      fs = tonumber(ass_fs_tag:match('%d+'))
+    end
+  end
+ 
+  settings.showamount = math.floor(playlist_h / fs)
+  
+  -- exclude the header line
+  if settings.playlist_header ~= "" then
+    settings.showamount = settings.showamount - 1
+  end
```

Comment:
I think here we should probably take newlines into account here.

Alternatively we could have a setting where you can define some N amount that is always subtracted from the auto showamount. That could be a bit easier since you could also adjust for different font sizes etc. But it does a little bit defeat the purpose of auto setting.

Reply:
`showamount`  subtract the number of `%N` or `\N` is very easy, but the height of empty line is much smaller than non empty line，which can leave too much blank space at the bottom.\
For Example `playlist_header=Playlist [%cursor/%plen]%N%N%N%N` ends with 4 empty lines, but the height is about 2 non empty lines.
![image](https://github.com/jonniek/mpv-playlistmanager/assets/45035465/8699d43a-3b2e-4131-b6e7-ee07e904e3cf)
I modified the `parse_header()` function to add an blank character to every `\N`, problem seems to be solved.
![image](https://github.com/jonniek/mpv-playlistmanager/assets/45035465/4327225e-4c36-4848-88d2-dc12b37a8b70)